web/views.py

```
@@ -6949,3 +6952,166 @@ def delete_post(request, post_id):
+            _request.session[f"recent_facts_{subject_id}"] = recent_facts[-5:]
+            _request.session.modified = True
+
+        except Exception:
+            logger = logging.getLogger(__name__)
+            logger.exception("Error saving fact for subject")
+            # Still return the generated fact even if we couldn't save it
+
+        return JsonResponse({"fact": fact})
+    except Subject.DoesNotExist:
+        return JsonResponse({"error": "Subject not found"}, status=404)
+    except Exception:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)